index.js

```diff
@@ -52,7 +52,7 @@ function ensureAllCommentsPrinted(astComments) {
 function format(text, opts, addAlignmentSize) {
   addAlignmentSize = addAlignmentSize || 0;
 
-  const ast = parser.parse(text, opts);
+  const ast = opts.astTransform(parser.parse(text, opts));
```

Comment:
I'd probably do this:

```js
opts.transform(
  parser,
  text,
  opts,
)
```

this enables the user to take in the text source and deal with parsing in the transformer. It also passes the default parser itself but the good thing is that this will enable third-party parsers by design.